packages/cli/src/external-secrets.ee/providers/aws-secrets-manager.ts

```diff
@@ -134,8 +171,60 @@ export class AwsSecretsManager implements SecretsProvider {
 	}
 
 	private assertAuthType(context: AwsSecretsManagerContext) {
-		if (context.settings.authMethod === 'iamUser') return;
+		const { authMethod } = context.settings;
+		if (authMethod === 'iamUser' || authMethod === 'autoDetect') return;
+		throw new UnknownAuthTypeError(authMethod);
+	}
+
+	private async fetchAllSecretsNames() {
+		const names: string[] = [];
+		let nextToken: string | undefined;
+
+		do {
+			const response = await this.client.listSecrets({
+				NextToken: nextToken,
+			});
 
-		throw new UnknownAuthTypeError(context.settings.authMethod);
+			if (response.SecretList) {
+				names.push(...response.SecretList.filter((s) => s.Name).map((s) => s.Name!));
+			}
+
+			nextToken = response.NextToken;
+		} while (nextToken);
+
+		return names;
+	}
+
+	private async fetchAllSecrets() {
+		const secrets: Secret[] = [];
+		const secretNames = await this.fetchAllSecretsNames();
+
+		// Batch the requests to avoid hitting limits
+		const batches = this.batch(secretNames);
+
+		for (const batch of batches) {
+			const response = await this.client.batchGetSecretValue({
+				SecretIdList: batch,
+			});
+
+			if (response.SecretValues) {
+				for (const secret of response.SecretValues) {
+					if (secret.Name && secret.SecretString) {
+						secrets.push({
+							secretName: secret.Name,
+							secretValue: secret.SecretString,
+						});
+					}
+				}
+			}
+		}
+
+		return secrets;
+	}
+
+	private batch<T>(arr: T[], size = 20): T[][] {
+		return Array.from({ length: Math.ceil(arr.length / size) }, (_, index) =>
+			arr.slice(index * size, (index + 1) * size),
+		);
```

Comment:
Since we're reusing instead of reimplementing we could use lodash instead:

https://github.com/n8n-io/n8n/blob/e6381e09e37dd40f7ba3222ec70b050fc08e0b06/packages/cli/src/active-workflow-manager.ts#L426